keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt

```diff
@@ -106,62 +106,77 @@ public class KeeperPlugin : Plugin<Project> {
       val appExtension = project.extensions.getByType<AppExtension>()
       val extension = project.extensions.create<KeeperExtension>("keeper")
-      val diagnosticOutputDir = project.layout.buildDirectory.dir("$INTERMEDIATES_DIR/diagnostics")
-      project.configureKeepRulesGeneration(appExtension, extension, diagnosticOutputDir)
-      project.configureL8Rules(appExtension, extension, diagnosticOutputDir)
+      project.configureKeepRulesGeneration(appExtension, extension)
+      project.configureL8Rules(appExtension, extension)
     }
   }
 
   private fun Project.configureL8Rules(
       appExtension: AppExtension,
-      extension: KeeperExtension,
-      diagnosticOutputDir: Provider<Directory>
+      extension: KeeperExtension
   ) {
     afterEvaluate {
       if (extension.enableL8RuleSharing.getOrElse(false)) {
-        val r8Enabled = !hasProperty("android.enableR8") ||
-            property("android.enableR8")?.toString()?.toBoolean() != false
-        if (r8Enabled) {
-          appExtension.onApplicableVariants(project, extension) { testVariant, appVariant ->
-            val appR8Task = "minify${appVariant.name.capitalize(Locale.US)}WithR8"
-            val androidTestL8Task = "l8DexDesugarLib${testVariant.name.capitalize(Locale.US)}"
-            val inputFiles = tasks
-                .named<R8Task>(appR8Task)
-                .flatMap { it.projectOutputKeepRules }
-
-            tasks
-                .named<L8DexDesugarLibTask>(androidTestL8Task)
-                .configure {
-                  keepRulesFiles.from(inputFiles)
-                  keepRulesConfigurations.set(listOf("-dontobfuscate"))
-
-                  if (extension.emitDebugInformation.getOrElse(false)) {
-                    doFirst {
-                      val mergedFilesContent = keepRulesFiles.files.asSequence()
-                          .flatMap { it.walkTopDown() }
-                          .filterNot { it.isDirectory }
-                          .joinToString("\n") {
-                            "# Source: ${it.absolutePath}\n${it.readText()}"
+        appExtension.onApplicableVariants(project, extension) { testVariant, appVariant ->
+          val appR8Task = "minify${appVariant.name.capitalize(Locale.US)}WithR8"
+          val androidTestL8Task = "l8DexDesugarLib${testVariant.name.capitalize(Locale.US)}"
+          val inputFiles = tasks
+              .named<R8Task>(appR8Task)
+              .flatMap { it.projectOutputKeepRules }
+
+          tasks
+              .named<L8DexDesugarLibTask>(androidTestL8Task)
+              .configure {
+                val taskName = name
+                keepRulesFiles.from(inputFiles)
+                keepRulesConfigurations.set(listOf("-dontobfuscate"))
+                val diagnosticOutputDir = layout.buildDirectory.dir(
+                    "$INTERMEDIATES_DIR/l8-diagnostics/$taskName")
+                    .forUseAtConfigurationTime()
+                    .get()
```

Comment:
Why do you use `.get().asFile` on this? Seems like it's more useful as a `DirectoryProperty`. On line 160 below, you could use that API to create a new file.

(My inner linter flags all usages of `java.io.File`)

Reply:
So if you try to resolve it as a directory property, gradle complains about it not existing :|. I can try again in a follow up but this was seemingly necessary

Reply:
:stare:

k, sounds good

Reply:
I wonder if something is eagerly triggering the configuration of the `L8DexDesugarLibTask`, at which point this file won't exist? 🤷